### ISSUE TITLE
Forked for bi-directional search

### ISSUE THREAD

**Issue by kohane27** [TheSafdarAwan/find-extender.nvim#8]
Hello @theSafdarAwan . Hope you're doing well.

I have forked your project which allows for bi-directional search (e.g., allowing pressing `f` or `t` to search the target in both directions, i.e., left and right at the same time.) Please see the [specific commit](https://github.com/kohane27/find-extender.nvim/commit/11f64e66cbfa6a5b30ed3cb2af71b0be7ce61da2). Essentially, I have just removed the code to check for direction.

I'm writing to see if you're still active on this project, and would like to accept a PR for this bi-directional search feature. We can add a flag in `config`. I'll then help with some housekeeping, like recording gif and formatting and stuff as discussed [here](https://github.com/theSafdarAwan/find-extender.nvim/discussions/7).

Thank you again for this great plugin. It completes my navigation in Neovim.

**Comment by theSafdarAwan** [comment]
> Hello @theSafdarAwan . Hope you're doing well.
> 
> I have forked your project which allows for bi-directional search (e.g., allowing pressing `f` or `t` to search the target in both directions, i.e., left and right at the same time.) Please see the [specific commit](https://github.com/kohane27/find-extender.nvim/commit/11f64e66cbfa6a5b30ed3cb2af71b0be7ce61da2). Essentially, I have just removed the code to check for direction.
> 
> I'm writing to see if you're still active on this project, and would like to accept a PR for this bi-directional search feature. We can add a flag in `config`. I'll then help with some housekeeping, like recording gif and formatting and stuff as discussed [here](https://github.com/theSafdarAwan/find-extender.nvim/discussions/7).
> 
> Thank you again for this great plugin. It completes my navigation in Neovim.

i had to leave programming all together for some time( i am not a professional programmer, just doing it for fun and someday might like to do some freelance projects) so for some personal problem i had to just take some time off of programming but i will definetenly continue this project.

**but why are you removing that code its an important bit to reverse the table because i parse the line then i have to manually make sure that the cursor lands on the right position**

Can you explain why, i don't understand your previous post on why to remove this.

**Comment by kohane27** [comment]
Hello @theSafdarAwan, thank you for getting back to me. I really appreciate it. Hopefully all is well with you.

It's reassuring that to know that you're not abandoning the project! I'm also excited to be working on this project with you: to make it better (as [discussed](https://github.com/theSafdarAwan/find-extender.nvim/discussions/7)).

> but why are you removing that code its an important bit to reverse the table because i parse the line then i have to manually make sure that the cursor lands on the right position

My apologies. I didn't take the time to understand the codebase properly and just hacked on it enough to achieve bi-directional search. I think it's because I don't need to use `F` (I only need `f`) so I don't need to reverse the table? I'm sorry for mishandling the code.


Once again, thank you for your dedication to this project:)

**Comment by kohane27** [comment]
My fork has a problem in which deleting, changing, visual selecting **backwards** doesn't work:

Deleting forwards like `dfte(label)` works:

```
this is a test for testing bi-directional search
^                  ^
cursor           target
```

Deleting backwards like the following is broken:

To delete target that is before the cursor (i.e., delete till `test`):

`dfte(label)`:

```
this is a test for testing bi-directional search
          ^                ^
       target            cursor
```

I think it's me messing up with the reversing the table when using `F` or `T`. Regardless, I'll look into it and get back asap. Thank you again

**Comment by theSafdarAwan** [comment]
> My fork has a problem in which deleting, changing, visual selecting **backwards** doesn't work:
> 
> Deleting forwards like `dfte(label)` works:
> 
> ```
> this is a test for testing bi-directional search
> ^                  ^
> cursor           target
> ```
> 
> Deleting backwards like the following is broken:
> 
> To delete target that is before the cursor (i.e., delete till `test`):
> 
> `dfte(label)`:
> 
> ```
> this is a test for testing bi-directional search
>           ^                ^
>        target            cursor
> ```
> 
> I think it's me messing up with the reversing the table when using `F` or `T`. Regardless, I'll look into it and get back asap. Thank you again

Sorry now that i read i think the thing that you were saying was if i have a cursor in the middle of the line you want to search the whole line.

By bi-directional is that your fork wasn't searching backward or forward.
Can you clear my misconception here. Thanks.

**Comment by kohane27** [comment]
> if i have a cursor in the middle of the line you want to search the whole line.

Yes! By bi-directional, I mean that when I press `f` or `t`, it'll search the whole line, regardless of the cursor location.

Please consider the following example:

```
this is a test for testing bi-directional search
          ^                ^
       target            cursor
```

Given the cursor is at `b` of `bi-directional`, I want to delete **backwards** `test for testing `. So I do `dftea`:

`d`: delete
`f`: find
`te` 2-key search for `test`
`a`: the label on screen

Expected result:

```
this is a bi-directional search
          ^
       cursor
```

Current buggy result:

```
this is a test for testing est for testing bi-directional search
                           ^
                         cursor
```

## More explanation

Currently, my fork works with jumping to location:

```
this is a test for testing bi-directional search
          ^                ^
       target            cursor
```

Pressing `ftea` works:

```
this is a test for testing bi-directional search
          ^
       cursor
```

But it won't work with operators like `deleting`, `changing` and visually selecting.

## Conclusion

I also understand that this "feature" is pretty niche. It's basically [Bidirectional search](https://github.com/ggandor/leap.nvim#faq) for the current line. I very much appreciate your willingness to entertain and look into this idea of mine. Thank you again!
